libraries/chain/include/graphene/chain/database.hpp

```diff
@@ -217,6 +217,21 @@ namespace graphene { namespace chain {
           */
          fc::signal<void(const vector<object_id_type>&, const vector<const object*>&, const flat_set<account_id_type>&)>  removed_objects;
 
+         /**
+          * Emitted after the calculation of the voting_stake (occurs in the maintenance interval)
+          */
+         fc::signal<void(const account_object&, const account_object&, const uint64_t)> on_voting_stake_calculated;
```

Comment:
At a glance I don't like it. Is there any way to avoid this?

Reply:
I need all the data to properly feed the plugin, so AFAIK there is no way to avoid this. Everything is performed inside the `vote_tally_helper` and nothing is given outside. 

What exactly do you don't like? Is the signal signature to specific? 

To make it more generic I could switch the emission signature to emit an object with the data. This would make it extensible. It would be up to the signal receiver to use the data or not.

Reply:
If I understood correctly, `void(const account_object&, const account_object&, const uint64_t)` means we need to emit the signal once per account, which IMHO is too many. I think it's easier if we add the `voting_stake` data directly into `account_statistics_object` as a member variable, although this would need a bit more memory.

Reply:
Yes you are right. The signal is emitted once per account. 

Inserting data in `account_statistics_object` would mean double tracking of the data. Also it would then insert always (or make a check whether the plugin is active, but this seems weird IMHO). But the plugin kinda looses its purpose then. 

What would be if I gathered the data first and then emit the signal with all the data? This would call the signal only once and wouldn't require more memory to be used. Though still would require a check whether the plugin is active or not.